### PR TITLE
filter speakers and submitters by presentation media uploads

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=3385040&stafilter=NotComplete&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Http/Utils/Filters/Filter.php [hunk 1]
```diff
@@ -634,4 +634,22 @@ public function getOriginalExp()
     {
         return $this->originalExp;
     }
+
+    /**
+     * @param FilterElement[] $e
+     * @return array
+     */
+    public static function getValueFromFilterElement(array $e): array {
```
Comment (smarcet) [inline app/Http/Utils/Filters/Filter.php:642]: i am no so sure that this should be a static method
IMHO should be an instance method using https://github.com/OpenStackweb/summit-api/blob/fcac0ee396cc8dbf287deef825f103ed04d29107/app/Http/Utils/Filters/Filter.php#L111

Comment (romanetar) [reply app/Http/Utils/Filters/Filter.php]: done

File: app/Models/Foundation/Summit/Speakers/PresentationSpeaker.php [hunk 2]
```diff
@@ -884,27 +884,27 @@ public function getAcceptedPresentations
 
         if (!is_null($filter)) {
             if ($filter->hasFilter("presentations_selection_plan_id")) {
-                $v = Filter::getValueFromFilterElement($filter->getFilter("presentations_selection_plan_id"));
+                $v = $filter->getValueFromFilterElement($filter->getFilter("presentations_selection_plan_id"));
```
Comment (smarcet) [inline app/Models/Foundation/Summit/Speakers/PresentationSpeaker.php:887]: use $this->getFilter inside getValueFromFilterElement to avoid this double call

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments

**Review by smarcet — APPROVED** [review]
LGTM

**Pull request merged**